eclipse-collections/src/main/java/org/eclipse/collections/impl/lazy/TakeWhileIterable.java

```diff
@@ -0,0 +1,133 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.impl.lazy;
+
+import java.util.Iterator;
+
+import org.eclipse.collections.api.block.predicate.Predicate;
+import org.eclipse.collections.api.block.procedure.Procedure;
+import org.eclipse.collections.api.block.procedure.Procedure2;
+import org.eclipse.collections.api.block.procedure.primitive.ObjectIntProcedure;
+import org.eclipse.collections.impl.lazy.iterator.TakeWhileIterator;
+
+/**
+ * Iterates over the elements of the adapted Iterable until the predicate returns false.
+ */
+public class TakeWhileIterable<T> extends AbstractLazyIterable<T>
+{
+    private final Iterable<T> adapted;
+    private final Predicate<? super T> predicate;
+
+    public TakeWhileIterable(Iterable<T> newAdapted, Predicate<? super T> predicate)
+    {
+        if (predicate == null)
+        {
+            throw new IllegalStateException("Predicate cannot be null");
+        }
+        this.adapted = newAdapted;
+        this.predicate = predicate;
+    }
+
+    public void each(Procedure<? super T> procedure)
+    {
+        boolean accept = true;
+        Iterator<T> iterator = this.adapted.iterator();
+        while (accept && iterator.hasNext())
+        {
+            T next = iterator.next();
+            if (this.predicate.accept(next))
+            {
+                procedure.value(next);
+            }
+            else
+            {
+                accept = false;
+            }
+        }
+    }
+
+    @Override
+    public void forEachWithIndex(ObjectIntProcedure<? super T> procedure)
+    {
+        int i = 0;
+        boolean accept = true;
+        Iterator<T> iterator = this.adapted.iterator();
+        while (accept && iterator.hasNext())
+        {
+            T next = iterator.next();
+            if (this.predicate.accept(next))
+            {
+                procedure.value(next, i);
+                i++;
+            }
+            else
+            {
+                accept = false;
+            }
+        }
+    }
+
+    @Override
+    public <P> void forEachWith(Procedure2<? super T, ? super P> procedure, P parameter)
+    {
+        boolean accept = true;
+        Iterator<T> iterator = this.adapted.iterator();
+        while (accept && iterator.hasNext())
+        {
+            T next = iterator.next();
+            if (this.predicate.accept(next))
+            {
+                procedure.value(next, parameter);
+            }
+            else
+            {
+                accept = false;
+            }
+        }
+    }
+
+    @Override
+    public Object[] toArray()
+    {
+        Object[] temp = new Object[this.size()];
```

Comment:
I think you should delete the override of toArray(). The class `TakeIterable` has an override of toArray() as an optimization, because we know that the size is likely to be `count`. In `TakeWhileIterable` we have no guess what the size will be, and we must always avoid calling `this.size()` on LazyIterables.